From 14fdafb78b4c8e1f3021ec7a7507b804dd7373c7 Mon Sep 17 00:00:00 2001
From: b07117 <b07117@freescale.com>
Date: Wed, 2 Nov 2011 18:26:33 +0800
Subject: [PATCH 200/635] ENGR00161253, disabling sencond UI during video
 playback

Read ro.secfb.disable-overlay to disable UI overlay

Signed-off-by: xianzhong <b07117@freescale.com>
---
 mx5x/hwcomposer/hwcomposer.cpp | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/mx5x/hwcomposer/hwcomposer.cpp b/mx5x/hwcomposer/hwcomposer.cpp
index 27770c9..8db2924 100755
--- a/mx5x/hwcomposer/hwcomposer.cpp
+++ b/mx5x/hwcomposer/hwcomposer.cpp
@@ -98,8 +98,15 @@ static int hwc_check_property(hwc_context_t *dev)
     }
     property_get("sys.SECOND_DISPLAY_ENABLED", value, "");
     if (strcmp(value, "1") == 0) {
-       property_set("sys.VIDEO_OVERLAY_DISPLAY", "2");
-       property_set("sys.VIDEO_DISPLAY", "0");
+       property_get("ro.secfb.disable-overlay", value, "0");
+       if (strcmp(value, "1") == 0){
+           property_set("sys.VIDEO_OVERLAY_DISPLAY", "0");
+           property_set("sys.VIDEO_DISPLAY", "1");
+       }
+       else{
+           property_set("sys.VIDEO_OVERLAY_DISPLAY", "2");
+           property_set("sys.VIDEO_DISPLAY", "0");
+       }
     } else
     {
        property_set("sys.VIDEO_OVERLAY_DISPLAY", "1");
-- 
1.8.0

